gapis/api/transform2/command_id.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package transform2 contains the elements to be able to transform
+// commands which consist of interfaces for individual transform operations
+// and a transform chain to run all of them.
+package transform2
+
+import (
+	"github.com/google/gapid/gapis/api"
+)
+
+type CommandType uint32
+
+const (
+	BeginCommand     CommandType = 0
+	TransformCommand CommandType = 1
+	EndCommand       CommandType = 2
+)
+
+type CommandID struct {
+	id          api.CmdID
+	commandType CommandType
+}
+
+func NewCommandID(id api.CmdID, cmdType CommandType) CommandID {
```

Comment:
I think we should have different constructors for the basic command types. NewBeforeCommands() NewRealCommand(uint32 id) NewAfterCommand()

As it is, you have to specify an id argument even when it doesn't make sense to for Begin and End.

Reply:
Makes sense.

Reply:
Only thing. I refrain to use "real" because it does not really implies what it means.